### DISCUSSION TITLE
Disclosure Hover States

### DISCUSSION THREAD

**Discussion by a-christensen** [twilio-labs/paste#835] [Show and tell]
Hey DSYS!

Thanks for your great ideas for the disclosure hover states. I explored the [two main options](https://share.goabstract.com/41ad32d6-7b55-45c1-8e80-8cb882a12dfb?mode=design&sha=bb28d7fc96e125e8143cdb73e9829b3dda95f38d) we talked about and would love to get your thoughts.

1. Button State - This version uses the "primary" blue as the text color (which is accessible) and adds a color-background-primary-lightest to make the hover state resemble our buttons. There is not currently a "primary" blue for text so we would need to discuss adding that to our styles if we decide this version is the best.
<img width="1093" alt="1_button_state" src="https://user-images.githubusercontent.com/52463764/95519552-8c596780-0982-11eb-9f4a-423f3e07f38d.png">

2. Arrow Move - This version explores having the arrow move right by $space-40 on hover. The thought is that the arrow would remain in this position on click while it transitions to a down arrow and only move back to its original position when the user mouses away. This would help reduce any animation weirdness. Check out a [rough demo](https://drive.google.com/file/d/1FfUU7PiUqdGmiuvr5dwiL-yG8vO4EtK8/view?usp=sharing) (note: design isn't exact).
<img width="1539" alt="2_arrow_move" src="https://user-images.githubusercontent.com/52463764/95519561-8fecee80-0982-11eb-8d8d-9679c69adaae.png">

Please take a look and add your thoughts, either casting a vote for one version or suggesting another approach/consideration. After we agree on a hover state, I can build out the remaining themes and examples for this component. 

Thanks so much for the help!

**Comment by richbachman** [comment]
Hi @a-christensen 

Thanks for putting these together! I'll share with the team for feedback.

**Reply by a-christensen** [reply]
Thanks, Rich! I really appreciate it.

**Comment by a-christensen** [comment]
I just added a [demo video](https://drive.google.com/file/d/1FfUU7PiUqdGmiuvr5dwiL-yG8vO4EtK8/view?usp=sharing) of the arrow animation so it's easier to conceptualize. Looking forward to hearing everyone's thoughts! :)

**Comment by kate-porter** [comment]
@a-christensen thanks so much for putting these options together! I LOVE both of these options.

What would you think of trying a blend of both? I feel like a text color change and the arrow animation could be really nice together. And the arrow animation feels maybe a liiittle dramatic - what would you think of moving it to the right by maybe $space-30 or $space-20?

Also, might I ask what you used to demo the animation? It's so cool!

**Reply by a-christensen** [reply]
Thanks for the feedback, Katie! I'm happy to try a blended version of both and include smaller spacing. My main concern was that when movement was the only indicator of hover, it could be easily missed if the disclosure was nearly full-width and the user was hovering on the opposite end of component. But the text color change would help a lot and so the movement wouldn't need to be so exaggerated. Thanks for the suggestions and I will add updates to this discussion.

I used Principle [https://principleformac.com/](url) to do the animation. I used that app back in the day when I was working on mobile designs.

**Reply by kate-porter** [reply]
Amazing! I played with principle awhile ago but haven't used it in awhile. I'm gonna have to play with it again as we start doing more animation design for Paste. :)

**Comment by a-christensen** [comment]
Check out some additional variations of the arrow animations.

1. [Arrow movement with $space-30](https://drive.google.com/file/d/1mPqjchp4QPvCVMX4dDXlR0dgpVp-IBQ_/view?usp=sharing)
2. [Arrow movement with $space-20](https://drive.google.com/file/d/1CYFbE8QlGw3triIln2j0IFwgvb4eE5xV/view?usp=sharing)
3. [Arrow movement with $space-20 and text color change to the primary blue](https://drive.google.com/file/d/1XtMXV1KBgLue-X43JjwpkYqyQGD00n4V/view?usp=sharing) (I'm open to other color suggestions here as well. We could go back to the link text color since we wouldn't need the color change to be accessible. I'm hesitant to do any gray options since we're using gray for the disabled state)

**Reply by a-christensen** [reply]
@katieporter Check these out when you get the chance and let me know your thoughts :)

**Reply by kate-porter** [reply]
Ohhh I like option #3! What do you think?

And yeah agreed on the blue text vs. gray. I think going back to the link text color would be a good option so we don't need to add a new token. The link text is the same color as the primary blue in the Default theme, anyway, so we'd be adding a new token that we'd really only need in the Console theme which doesn't feel worth it.

**Comment by corinne-nagel** [comment]
OMG @a-christensen Love option 3, this is looking so great!

**Comment by a-christensen** [comment]
Yayyy! Thanks for the feedback, crew. Here's an updated version of [$space-20 w/ color change to link text](https://drive.google.com/file/d/15X9y3f0CNdqFyaoLGDPYM13lXM1kZzBz/view?usp=sharing) for comparison. This might be the best so we don't have to create any new text styles.

**Reply by kate-porter** [reply]
Love it! 🙌
